torchgeo/datasets/sentinel.py

```diff
@@ -25,6 +25,185 @@ class Sentinel(RasterDataset):
     """
 
 
+class Sentinel1(Sentinel):
+    """Sentinel-1 dataset.
+
+    The `Sentinel-1 mission
+    <https://sentinel.esa.int/web/sentinel/missions/sentinel-1>`_ comprises a
+    constellation of two polar-orbiting satellites, operating day and night
+    performing C-band synthetic aperture radar imaging, enabling them to
+    acquire imagery regardless of the weather.
+
+    Data can be downloaded from:
+
+    * `Copernicus Open Access Hub
+      <https://scihub.copernicus.eu/>`_
+    * `Alaska Satellite Facility (ASF) Distributed Active Archive Center (DAAC)
+      <https://asf.alaska.edu/>`_
+    * `Microsoft's Planetary Computer
+      <https://planetarycomputer.microsoft.com/dataset/sentinel-1-rtc>`_
+
+    Product Types:
+
+    * `Level-0
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-0>`_:
+      Raw (RAW)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Single Look Complex (SLC)
+    * `Level-1
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-1>`_:
+      Ground Range Detected (GRD)
+    * `Level-2
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/product-types-processing-levels/level-2>`_:
+      Ocean (OCN)
+
+    Polarizations:
+
+    * HH: horizontal transmit, horizontal receive
+    * HV: horizontal transmit, vertical receive
+    * VV: vertical transmit, vertical receive
+    * VH: vertical transmit, horizontal receive
+
+    Acquisition Modes:
+
+    * `Stripmap (SM)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/stripmap>`_
+    * `Interferometric Wide (IW) swath
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/interferometric-wide-swath>`_
+    * `Extra Wide (EW) swatch
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/extra-wide-swath>`_
+    * `Wave (WV)
+      <https://sentinels.copernicus.eu/web/sentinel/user-guides/sentinel-1-sar/acquisition-modes/wave>`_
+
+    At the moment, this dataset only supports the GRD product type. Data must be
+    radiometrically terrain corrected (RTC). This can be done manually using a DEM, or
+    you can download an On Demand RTC product from ASF DAAC.
```

Comment:
If we do add support for SLC/OCN/RAW, there's a good chance we'll need separate classes to support these. I'll leave this for future work and stick with `Sentinel1` for now.